impeller/entity/shaders/blending/framebuffer_blend.frag

```diff
@@ -43,14 +43,25 @@ vec4 Sample(sampler2D texture_sampler, vec2 texture_coords) {
 }
 
 void main() {
-  f16vec4 dst = IPHalfUnpremultiply(f16vec4(ReadDestination()));
-  f16vec4 src = IPHalfUnpremultiply(
+  f16vec4 premultiplied_dst = f16vec4(ReadDestination());
+  f16vec4 premultiplied_src =
       f16vec4(Sample(texture_sampler_src,  // sampler
                      v_src_texture_coords  // texture coordinates
-                     )));
-  src.a *= frag_info.src_input_alpha;
+                     ));
+  int nblend_type = int(blend_type);
 
-  f16vec3 blend_result = AdvancedBlend(dst.rgb, src.rgb, int(blend_type));
+  if (nblend_type == /*BlendSelectValues::kPlusAdvanced*/ 14) {
+    f16vec4 plus = premultiplied_dst + premultiplied_src;
+    if (plus.a > 1.0hf) {
+      plus.a = 1.0hf;
+    }
```

Comment:
nit: use clamp

Reply:
I'm trying, I can't get it to compile for iOS because of our half float trickery.

Reply:
🤦

Reply:
Its generally cheap to convert float -> half and back, that said its a really minor nit. clamp is probably doing the same thing.

Reply:
My understanding is that it isn't the same thing, that the hardware has specific accommodations for clamping.  It should be faster, especially when you care about both sides of the clamp.  In the early days of shaders branching wasn't even allowed and you had to use `clamp`.  That my not be as big of a deal on modern hardware though.

Reply:
I finally got it, this works
```
f16vec4 IPHalfPlusBlend(f16vec4 src, f16vec4 dst) {
  float16_t min = 0.0hf;
  float16_t max = 1.0hf;
  return clamp(dst + src, min, max);
}
```
This doesn't
```
f16vec4 IPHalfPlusBlend(f16vec4 src, f16vec4 dst) {
  return clamp(dst + src, 0.0hf, 1.0hf);
}
```

Reply:
I think its not as much anymore, SIMT vs SIMD